### PR TITLE
Add recipe to add gradle enterprise maven extension to maven projects

### PR DESCRIPTION
**Pull request by kunli2**
closes https://github.com/openrewrite/rewrite/issues/3005

To implement the required recipe [here](https://github.com/openrewrite/rewrite/issues/3005) 

To integrate Gradle Enterprise Maven Extension into Maven projects, ensure that the `gradle-enterprise-maven-extension` is added to the `.mvn/extensions.xml` file if not already present. 
Additionally, configure the extension by adding the `.mvn/gradle-enterprise.xml` configuration file.

### CODE REVIEW

File: rewrite-maven/src/main/java/org/openrewrite/maven/AddGradleEnterpriseMavenExtension.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+    @Option(displayName = "Plugin version",
+        description = "An exact version number or node-style semver selector used to select the gradle-enterprise-maven-extension version.",
+        example = "3.x")
+    @Nullable
+    String version;
+
+    @Option(displayName = "Server URL",
+        description = "The URL of the Gradle Enterprise server.",
+        example = "https://ge.openrewrite.org/")
```
Comment (sambsnyd) [inline rewrite-maven/src/main/java/org/openrewrite/maven/AddGradleEnterpriseMavenExtension.java:69]:
```suggestion
        example = "https://scans.gradle.com/")
```

File: rewrite-maven/src/main/java/org/openrewrite/maven/AddGradleEnterpriseMavenExtension.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+                                         "    xmlns=\"https://www.gradle.com/gradle-enterprise-maven\" " +
+                                         "xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\n" +
+                                         "    xsi:schemaLocation=\"https://www.gradle.com/gradle-enterprise-maven" +
+                                         " https://www.gradle.com/schema/gradle-enterprise-maven.xsd\">\n" +
+                                         "  <server>\n" +
+                                         "    <url>%s</url>\n" +
+                                         "    <allowUntrusted>%b</allowUntrusted>\n" +
+                                         "  </server>\n" +
+                                         "  <buildScan>\n" +
+                                         "    <backgroundBuildScanUpload>false</backgroundBuildScanUpload>\n" +
```
Comment (sambsnyd) [inline rewrite-maven/src/main/java/org/openrewrite/maven/AddGradleEnterpriseMavenExtension.java:56]: Why is this hardcoded to `false`? Is this parameter required? If we're not making it configurable would rather not set a value for it unless required

**Review by sambsnyd — APPROVED** [review]

File: rewrite-maven/src/main/java/org/openrewrite/maven/AddGradleEnterpriseMavenExtension.java (hunk 2 above)
Comment (kunli2) [reply rewrite-maven/src/main/java/org/openrewrite/maven/AddGradleEnterpriseMavenExtension.java:71]: Yes, good question, I thought it was false by default, just asked the question [here](https://github.com/openrewrite/rewrite/issues/3005#issuecomment-1533593785)

Comment (kunli2) [reply rewrite-maven/src/main/java/org/openrewrite/maven/AddGradleEnterpriseMavenExtension.java:71]: I saw @alextu is OOO until May 8, so maybe let us keep it as is (not sure if it's a required field and safe to remove it), and we can make another code update if we want to make it configurable.

**Pull request merged**